crates/database/src/solver_competition.rs

```diff
@@ -1,26 +1,16 @@
 use crate::{auction::AuctionId, TransactionHash};
 use sqlx::{types::JsonValue, PgConnection};
 
-// TODO: once the auction_transaction are verified to work we can remove the tx_hash
-// column and argument.
 pub async fn save(
     ex: &mut PgConnection,
     id: AuctionId,
     data: &JsonValue,
-    tx_hash: Option<&TransactionHash>,
 ) -> Result<(), sqlx::Error> {
     const QUERY: &str = r#"
-INSERT INTO solver_competitions (id, json, tx_hash)
-VALUES ($1, $2, $3)
-ON CONFLICT (id) DO UPDATE
-SET json = EXCLUDED.json, tx_hash = EXCLUDED.tx_hash
+INSERT INTO solver_competitions (id, json)
+VALUES ($1, $2)
```

Comment:
This specifically prevents multiple `solver_competitions` rows for the same auction ID if I understand correctly.

Reply:
Yes. This prevents it. Also, inserting the id again into auction_transaction would fail. Both happen when the competition info is stored in a transaction so either one failing prevents it (as mentioned on Slack).